src/staking/block_validator.cpp

```diff
@@ -178,9 +179,9 @@ class BlockValidatorImpl : public AbstractBlockValidator {
       uint256 *snapshot_hash_out       //!< [out] The snapshot hash extracted from the scriptSig
       ) const override {
 
-    // check that there are transactions
-    if (block.vtx.empty()) {
-      return BlockValidationResult(Error::NO_TRANSACTIONS);
+    // check block size limits
+    if (!CheckBlockWeight(block)) {
```

Comment:
nit: is a bit inconsistent imho to have `CheckBlockWeight()` but then return a `INVALID_BLOCK_LENGTH`